### PR TITLE
Don't try to fix resources location when --resources is used.

### PR DESCRIPTION
**Pull request by thomasballinger**
**Bugfix:** This PR resolves https://github.com/endless-sky/endless-sky/issues/6220

## Fix Details
Only apply most adjustments to the resources path when it has not been explicitly specified on the command line.

Still `FixWindowsSlashes(resources)` and append a trailing slash if it is missing.

## Testing Done
I tested the new mac behavior against the repro described in https://github.com/endless-sky/endless-sky/issues/6220 and a credits.txt file in the parent directly no longer causes resource loading to fail. I have not tested the linux changes.

### CODE REVIEW

File: source/Files.cpp [hunk 1]
```diff
@@ -91,28 +95,32 @@ void Files::Init(const char * const *argv)
...
 #endif
+	}
 	// If the resources are not here, search in the directories containing this
 	// one. This allows, for example, a Mac app that does not actually have the
 	// resources embedded within it.
 	while(!Exists(resources + "credits.txt"))
 	{
 		size_t pos = resources.rfind('/', resources.length() - 2);
-		if(pos == string::npos || pos == 0)
+		if(pos == string::npos || pos == 0 || resourcesSpecifiedWithArg)
```
Comment (tehhowch) [inline source/Files.cpp:123]: Mind splitting the Init implementation up, e.g. into private / namespaced helper methods `SetResourceDirectory(string&,bool` and `SetConfigDirectory(string&)`?

Then we can immediately bail if a resource or config dir is specified but invalid, with a more meaningful runtime_error message.

File: source/Files.cpp [hunk 2]
```diff
@@ -57,109 +57,123 @@ namespace {
...
-		char *str = SDL_GetBasePath();
-		if(!str)
-			throw runtime_error("Unable to get path to resource directory!");
-		
-		resources = str;
-		SDL_free(str);
-	}
+		bool specifiedWithArg = resources.empty();
+		if(resources.empty())
+		{
```
Comment (tehhowch) [inline source/Files.cpp:65]: My earlier comment was more to the tune of
```cpp
if(!resources.empty())
{
  // Add / if needed
  if(!Files::Exists(resources + "credits.txt"))
    throw runtime_error("provided resources directory \"" + resources + "\" is not valid");
}
else
{
  // Go looking.
}
// Verify data, images, sounds, etc.
```

File: source/Files.cpp [hunk 3]
```diff
@@ -57,109 +57,123 @@ namespace {
...
+			SDL_free(str);
+
+			// Create the "plugins" directory if it does not yet exist, so that it is
+			// clear to the user where plugins should go.
+			{
+				char *str = SDL_GetPrefPath("endless-sky", "plugins");
+				if(str != nullptr)
+					SDL_free(str);
+			}
+		}
```
Comment (tehhowch) [inline source/Files.cpp:82]: This should probably still happen even if the game is always started with a resource path.

(It seems more suited to config, as well)

Comment (thomasballinger) [reply source/Files.cpp:82]: Are you thinking create plugins directory wherever config is set to (my preference) or always at this default location? re moving to config, whoops, yep will do.

Comment (tehhowch) [reply source/Files.cpp:82]: If a config directory is provided, then yes, we'd want to ensure the expected directory structure exists in it.

Comment (thomasballinger) [reply source/Files.cpp:82]: Ok, so by default we'll use SDL_GetPrefPath("endless-sky", "plugins"); to create the directory, but if a custom config is specified we'll check for plugins and create a folder there if necessary using system specific calls.

**Comment by thomasballinger** [comment]
The pattern is now

```
if (resources.empty())
	SetResourceDirectory(string &resources) // go looking
else
{
	// fix up the string (windows \ -> /, add a trailing /)
}
// throw runtime error if resources do not exist
```
however credits.txt for resources is not longer enforced explicitly set directories, it's only used for discovery.

Does this pattern work, or would you prefer I swap the conditional to `if (!resources.empty()`?

Should credits.txt in resources be required when `--resources` is manually specified? (currently only data, images, and sounds are required)

File: source/Files.cpp [hunk 4]
```diff
@@ -58,70 +58,60 @@ namespace {
...
+
 #if defined _WIN32
-	FixWindowsSlashes(resources);
+		FixWindowsSlashes(resources);
 #endif
-	if(resources.back() != '/')
-		resources += '/';
+		if(resources.back() != '/')
+			resources += '/';
```
Comment (tehhowch) [inline source/Files.cpp:85]: This is a pattern repeated a couple times (fixing slashes if windows and then ensuring the string ends in `/`), mind extracting to a helper function?

**Review by tehhowch — COMMENTED** [review]
I like to think of the check for credits.txt in the resources directory as a "you must acknowledge those who built the content / engine when using it," but it's really not a robust check at all.

**Comment by quyykk** [comment]
#7929

**Pull request closed without merging**